### PR TITLE
187 - Optimize pytorch weight loading on startup + dependencies update

### LINKED ISSUE #187: [Bug]: Fix VisionNode delay in leaderboard
### Current Behavior

VisionNode-Output is delayed by multiple seconds in the leaderboard.

### Expected Behavior

Output roughly in sync with camera output.

### How to reproduce the issue

* launch leaderboard and check rviz

### PR DESCRIPTION
**Pull request by MaxJa4**
# Description

Fixes PyTorch auto-downloading models on node-start even if its not used (causes significant delay until ready to process).
Dependencies updated to fix building issues (b5 update) and have newest versions of PyTorch/Ultralytics and other libraries.

Fixes #187 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by MaxJa4** [comment]
Delay reduced to ~0.5s on my machine. Won't be at university until Thursday due to work - would appreciate a quick test of this fix on the lab PCs to see if it works :) Thanks

**Comment by samuelkuehnel** [comment]
Here is a short video from the delay, it is about 1-2 seconds when looking at the ros time.
[delay.webm](https://github.com/una-auxme/paf23/assets/51356601/7f727aaa-cc45-435d-bcee-c37ab4a3c0ce)

**Comment by MaxJa4** [comment]
@samuelkuehnel Thanks for testing. I saw that with the recent additions in the VisionNode by the LIDAR/distance features by Leon, the overall calculation time increased. It's ~50ms (20 FPS) on my PC, looks like it's more like 200ms on the lab PCs. I can try to check if I can do anything to fix this in the VisionNode without breaking the distance/LIDAR stuff, but I doubt it.

**Comment by MaxJa4** [comment]
Added further measures to ensure up-to-date classification data:
- Traffic light detection is handled asynchronously now (-> independent of distance calculation)
- Switch to second best classification model (YOLO RTDETR-L instead of RTDETR-X) which ~54% faster with only slightly worse accuracy according to [Ultralytics](https://docs.ultralytics.com/models/rtdetr/#pre-trained-models)

Overall, the timings of the VisionNode changed like this (tested on a 4070 Ti):

Before:
- Classification: ~50ms
- Traffic light detection & distance calc: ~22ms
- Total: ~72ms -> 13.9 FPS

Now:
- Classification: ~40ms
- Traffic light detection: asynchronously now, doesn't wait for distance calc
- Distance calc: ~15ms
- Total: ~60ms -> 16.7 FPS

Using the NAS models was not possible because of (hours of) dependency hell with Ultralytics + Super-Gradients... also converting the RT-DETR model to a way faster format like TensorRT is not loadable with Ultralytics.

**Dependencies changed, please run `b5 update`! :)**

**Comment by MaxJa4** [comment]
5 days ago, a new/improved version of the standard YOLO model has been released: https://github.com/WongKinYiu/yolov9
This may be a candidate to try if the above changes are not sufficient performance wise.

**Review by samuelkuehnel — APPROVED** [review]
